brave/brave-core
# Add additional Rewards metrics

Resolves https://github.com/brave/brave-browser/issues/38567



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

Start with fresh profile, ensure no metrics are present. Enable Rewards.

For `Brave.Rewards.AdTypesEnabled.2`:

0. Ensure initial metric value is 7.
1. Toggle some ad types, ensure metric value is correct.

For `Brave.Rewards.SearchResultAdsOptin`

0. Turn off & turn on search ads. Ensure metric does not exist
1. Turn off search ads and connect a payout account. Ensure metric does not exist
2. Turn on search ads, ensure metric exists with value 1.

For `Brave.Rewards.AdsHistoryView`:

View ads history, ensure metric has value 1.
